### PR TITLE
Add if-match condition to PutAppendBlob

### PR DESCRIPTION
**Pull request by ishai42**

### CODE REVIEW

File: sdk/storage_blobs/src/blob/operations/put_append_blob.rs
Comment (heaths) [inline sdk/storage_blobs/src/blob/operations/put_append_blob.rs:35]: We tend to do `If-Match` and `If-None-Match` headers in a pair. Could you add support for `If-None-Match` as well?

Comment (ishai42) [reply sdk/storage_blobs/src/blob/operations/put_append_blob.rs:35]: I believe these are handled by the match condition meaning the following fragment from the test generates if-none-match header:

```
    let result = blob
        .put_append_blob()
        .content_type("text/plain")
        .if_match(IfMatchCondition::NotMatch("*".into()))
        .await
        .unwrap_err();
```

Comment (ishai42) [reply sdk/storage_blobs/src/blob/operations/put_append_blob.rs:37]: I guess you mean if-modified-since.   Added.

Comment (heaths) [reply sdk/storage_blobs/src/blob/operations/put_append_blob.rs:37]: I meant that header, but didn't realize that's how it was being handled. If that's the case, then this code looked fine.

**Review by heaths — APPROVED** [review]
The added condition doesn't hurt, though.

**Comment by heaths** [comment]
I'm working on fixing the new, breaking nightly clippy rule violations. At least they have fixes available.

**Comment by heaths** [comment]
Blocked on #1667; I will rebase after that merges.

**Pull request merged**
